pkgs/applications/science/logic/isabelle/default.nix

```diff
@@ -1,6 +1,38 @@
-{ lib, stdenv, fetchurl, coreutils, nettools, java, scala, polyml, z3, veriT, vampire, eprover-ho, naproche, rlwrap, perl, makeDesktopItem, isabelle-components, isabelle, symlinkJoin }:
+{ lib, stdenv, fetchurl, coreutils, nettools, java, scala, polyml, z3, veriT, vampire, eprover-ho, naproche, rlwrap, perl, makeDesktopItem, isabelle-components, isabelle, symlinkJoin, fetchhg, runCommand }:
 # nettools needed for hostname
 
+let
+  sha1 = stdenv.mkDerivation {
+    name = "isabelle-sha1";
+
+    src = fetchhg {
+      url = "https://isabelle.sketis.net/repos/sha1";
+      rev = "e0239faa6f42";
+      sha256 = "sha256-4sxHzU/ixMAkSo67FiE6/ZqWJq9Nb9OMNhMoXH2bEy4=";
+    };
+
+    buildPhase = (if stdenv.isDarwin then ''
+      LDFLAGS="-dynamic -undefined dynamic_lookup -lSystem"
+    '' else ''
+      LDFLAGS="-fPIC -shared"
+    '') + ''
+      CFLAGS="-fPIC -I."
+      $CC $CFLAGS -c sha1.c -o sha1.o
+      $LD $LDFLAGS sha1.o -o libsha1.so
+    '';
+
+    installPhase = ''
+      mkdir -p $out/lib
+      cp libsha1.so $out/lib/libsha1.so
+    '';
+  };
+  ml_home = runCommand "ml-home" { } ''
+    mkdir $out
+    ln -s ${polyml}/bin/* $out
+    ln -s ${polyml}/lib/* $out
+    ln -s ${sha1}/lib/* $out
```

Comment:
Does this even end up in someones configuration? If so we can't link thing to /.

Reply:
Sorry, I don't understand what you mean by "can't link thing to /".

Reply:
@SuperSandro2000 could you clarify what you mean by this?

Reply:
I think he meant that libraries should go to `$out/lib` and binaries to `$out/bin`. Could you change that and see if it still works?

Reply:
Oh, I don't think that will work, since I'm pretty sure Isabelle wants libsha1.so in the same directory as the poly binary.

Reply:
I have a patch that fixes that. Is it ok if I take over here? I won't be able to push to your branch, so I would have to open a new PR

Reply:
If you want to amend your PR, here is the code: https://github.com/jvanbruegge/nixpkgs/commit/9aa4c5dbfc806b2f71ad351fc79d5d0312e0caa4

Reply:
I've amended my PR.